File: LeetCode_Scala/src/com/rp/leetcode/LeetCode7.scala
package com.rp.leetcode

object LeetCode7 {
  def reverse(x: Int): Int = {
    var str = x.toString()
    if (x < 0) {
      str = str.substring(1)
      str = str.reverse
      if (-1 * str.toLong < Int.MinValue) {
        return 0
      }
      return -1 * str.toInt
    }
    str = str.reverse
    if (str.toLong > Int.MaxValue)
      0
    else
      str.toInt
  }

  def easy(x: Int): Int = {
    var ans: Long = 0
    var yu: Int = 0
    var temp: Int = x
    while (temp != 0) {
      yu = temp % 10
      temp /= 10
      if (ans > Integer.MAX_VALUE / 10 || ans < Integer.MIN_VALUE / 10)
        return 0
      ans = ans * 10 + yu
    }
    ans.toInt
  }

  def fastest(x: Int): Int = {
    if (x == 0) x
    else{
      try{
        val absX = Math.abs(x)
        x / absX * absX.toString.reverse.toInt
      }catch {
        case _: Throwable => 0
      }
    }
  }

  def main(args: Array[String]): Unit = {
    val arr = Array(123, -123, 120, 10, 969, 898, 0, 1534236469)
    for (i <- 0 to (arr.length - 1)) {
      println(reverse(arr(i)))
      println(easy(arr(i)))
      println(fastest(arr(i)))
    }
  }
}

File: LeetCode_Java/src/com/leetcode/rp/LeetCode7.java
package com.leetcode.rp;

class SolutionOfLeetCode7 {
    public int reverse(int x) {
        int flag = 1;
        String string;
        if (x < 0) {
            flag = -1;
            string = x + "";
            string = string.substring(1, string.length());
        } else {
            string = x + "";
        }
        String newstring = new String();
        int length = string.length();
        for (int i = length - 1; i > -1; i--) {
            newstring += string.charAt(i);
        }
        try {
            x = Integer.parseInt(newstring);
        } catch (NumberFormatException e) {
            return 0;
        }
        return x * flag;
    }

    public int easy(int x) {
        int ans = 0;
        int yu = 0;
        while (x != 0) {
            yu = x % 10;
            x = x / 10;
            if (ans > Integer.MAX_VALUE / 10 || ans < Integer.MIN_VALUE / 10)
                return 0;
            ans = ans * 10 + yu;
        }
        return ans;
    }
}

public class LeetCode7 {
    public static void main(String[] args) {
        SolutionOfLeetCode7 solutionOfLeetCode7 = new SolutionOfLeetCode7();
        int[] x = {123, -123, 120, 21, 1534236469, -2147483648};
        for (int i : x) {
            int ans = solutionOfLeetCode7.reverse(i);
            System.out.println(ans);
            System.out.println(solutionOfLeetCode7.easy(i));
        }
    }
}
